components/public-api/gitpod/experimental/v1/oidc.proto

```diff
@@ -0,0 +1,146 @@
+
+  // URL using the https scheme with no query or fragment component that the OP
+  // asserts as its Issuer Identifier. Required.
+  string issuer = 2;
+
+  // A KeySet that can validate the id_token (JSON web token)
+  // Either one is required.
+  string jwks = 3;
+  string jwks_url = 4;
+
+  // Control options for the consent screen.
+  // Optional.
+  string prompt = 5;
+  // A hint to pre-select the tenant from an AD.
+  // Optional.
+  string domain_hint = 6;
+  // Optional.
+  string login_hint = 7;
+
+  // If set, the default claim mapping is overriden.
+  // Optional.
+  bool override_claim_mapping = 8;
+  // Optional.
+  string claim_email_key = 9;
+  // Optional.
+  string claim_groups_key = 10;
+  // Optional.
+  string claim_username_key = 11;
+  // END – OIDC
+
+  // BEGIN – OAuth2 configuration
+  // Required.
+  string client_id = 12;
+  // Required.
+  string client_secret = 13;
+  // Required.
+  string authorization_endpoint = 14;
+  // Required.
+  string token_endpoint = 15;
+  // Required.
+  repeated string scopes_supported = 16;
+
+  // Source for additional claims for the token.
+  // Required.
+  string userinfo_endpoint = 17;
+  // Optional.
+  string userinfo_id_key = 18;
+  // Optional.
+  string userinfo_name_key = 19;
+
+  // Based on this flag, the provider is not using OpenID for authentication,
+  // the request_type will be "code" instead of "id_token", and the
+  // "userinfo_endpoint" will be queried to obtain a profile by default.
+  bool is_oauth_only = 20;
+  // END – OAuth2 configuration
+
+  // List of the JWS signing algorithms (alg values) supported by the OP for the
+  // ID Token to encode the Claims in a JWT. The algorithm RS256 MUST be
+  // included. Optional.
+  repeated string id_token_signing_alg_values_supported = 21;
+
+  // Time when the config was created.
+  // Read-only.
+  google.protobuf.Timestamp creation_time = 22;
+}
+
+service OIDCProviderConfigService {
+  // Creates a new OIDC provider configuration.
+  rpc CreateOIDCProviderConfig(CreateOIDCProviderConfigRequest)
+      returns (CreateOIDCProviderConfigResponse) {};
+
+  // Retrieves an OIDC provider configuration.
+  rpc GetOIDCProviderConfig(GetOIDCProviderConfigRequest)
+      returns (GetOIDCProviderConfigResponse) {};
+
+  // Lists OIDC provider configurations.
+  rpc ListOIDCProviderConfigs(ListOIDCProviderConfigsRequest)
+      returns (ListOIDCProviderConfigsResponse) {};
+
+  // Updates modifiable properties of an existing OIDC provider configuration.
+  rpc UpdateOIDCProviderConfig(UpdateOIDCProviderConfigRequest)
+      returns (UpdateOIDCProviderConfigResponse) {};
+
+  // Removes a OIDC provider configuration by ID.
+  rpc DeleteOIDCProviderConfig(DeleteOIDCProviderConfigRequest)
+      returns (DeleteOIDCProviderConfigResponse) {};
+}
+
+message CreateOIDCProviderConfigRequest {
+  OIDCProviderConfig config = 1;
+
+  // Optional.
+  bool use_discovery = 2;
+  // Optional.
+  bool is_oauth_only = 3;
```

Comment:
```suggestion
  bool oauth_only = 3;
```
The boolean type already indicates "the questions you can ask against this field". See https://google.aip.dev/140